### PR TITLE
Fixing incorrect asserts in CXformUtils

### PR DESCRIPTION
**Pull request by vraghavan78**
This is a legacy bug, an incorrect assert. Adding other operators in the
in the assert.

### CODE REVIEW

File: libgpopt/include/gpopt/xforms/CXformUtils.inl [hunk 1]
```diff
@@ -73,7 +73,9 @@ namespace gpopt
 			COperator::EopPhysicalInnerNLJoin ==  eopid ||
 			COperator::EopPhysicalLeftOuterNLJoin ==  eopid ||
 			COperator::EopPhysicalLeftSemiNLJoin ==  eopid ||
-			COperator::EopPhysicalLeftAntiSemiNLJoin
+			COperator::EopPhysicalLeftAntiSemiNLJoin  ==  eopid ||
+			COperator::EopPhysicalLeftAntiSemiNLJoinNotIn ==  eopid ||
+			COperator::EopPhysicalLeftOuterNLJoin ==  eopid
```
Comment (hsyuan) [inline libgpopt/include/gpopt/xforms/CXformUtils.inl]: we already had this in L74

Comment (vraghavan78) [reply libgpopt/include/gpopt/xforms/CXformUtils.inl]: sorry will remove that

**Comment by d** [comment]
@vraghavan78 I reworded the commit and also removed a duplicate.

File: libgpopt/include/gpopt/xforms/CXformUtils.inl [hunk 2]
```diff
@@ -73,7 +73,8 @@ namespace gpopt
 			COperator::EopPhysicalInnerNLJoin ==  eopid ||
 			COperator::EopPhysicalLeftOuterNLJoin ==  eopid ||
 			COperator::EopPhysicalLeftSemiNLJoin ==  eopid ||
-			COperator::EopPhysicalLeftAntiSemiNLJoin
+			COperator::EopPhysicalLeftAntiSemiNLJoin  ==  eopid ||
+			COperator::EopPhysicalLeftAntiSemiNLJoinNotIn ==  eopid
```
Comment (hsyuan) [inline libgpopt/include/gpopt/xforms/CXformUtils.inl]: there are 2 spaces after `==`

Comment (d) [reply libgpopt/include/gpopt/xforms/CXformUtils.inl]: Good catch!

File: libgpopt/include/gpopt/xforms/CXformUtils.inl [hunk 3]
```diff
@@ -70,10 +70,11 @@ namespace gpopt
...
 		GPOS_ASSERT
 			(
-			COperator::EopPhysicalInnerNLJoin ==  eopid ||
-			COperator::EopPhysicalLeftOuterNLJoin ==  eopid ||
-			COperator::EopPhysicalLeftSemiNLJoin ==  eopid ||
-			COperator::EopPhysicalLeftAntiSemiNLJoin
+			COperator::EopPhysicalInnerNLJoin == eopid ||
+			COperator::EopPhysicalLeftOuterNLJoin == eopid ||
+			COperator::EopPhysicalLeftSemiNLJoin == eopid ||
+			COperator::EopPhysicalLeftAntiSemiNLJoin  == eopid ||
```
Comment (hsyuan) [inline libgpopt/include/gpopt/xforms/CXformUtils.inl]: 1 more space before `==` ;)

**Pull request merged**